Configure.cmake

```diff
@@ -42,6 +45,28 @@ CHECK_C_SOURCE_COMPILES("
     __m128d r = _mm_mul_pd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
   COMPILER_SUPPORTS_SSE2)
 
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_SSE4})
+CHECK_C_SOURCE_COMPILES("
+  #if defined(_MSC_VER)
+  #include <intrin.h>
+  #else
+  #include <x86intrin.h>
+  #endif
+  int main() {
+    __m128d r = _mm_floor_sd(_mm_set1_pd(1), _mm_set1_pd(2)); }"
+  COMPILER_SUPPORTS_SSE4)
+
+set (CMAKE_REQUIRED_FLAGS ${FLAGS_ENABLE_AVX})
```

Comment:
Can we leave this as it was originally? I don't like the solution of using `CHECK_C_SOURCE_RUNS` to set `COMPILER_SUPPORT_*` as much as I don't like checking if the compiler is Cygwin.

If we use the original version to use `CHECK_C_SOURCE_COMPILES` to set `CMAKE_SUPPORTS_AVX` on Cygwin, we would end up building a library that provides symbols (the AVX ones) that the system (Cygwin) is not able to run? Is that a bad thing? @shibatch what do you think? I don't see why this is a bad thing, as we are already building the library for AVX512  on linux x86 even if the target system does not support it. Am I wrong?